### PR TITLE
New test cases for `fill_nulls_with`

### PR DESCRIPTION
**Pull request by courtneyholcomb**
### Description
Tests for some edge cases that have come up in discussions about coalescing nulls. Getting these SQL examples merged will allow us to see the SQL changes more clearly when working on follow ups like [SL-1677](https://linear.app/dbt-labs/issue/SL-1677/coalesce-nulls-is-not-being-applied-to-conversion-metric-input).

### CODE REVIEW

File: metricflow/test/fixtures/semantic_manifest_yamls/simple_manifest/metrics.yaml [hunk 1]
```diff
@@ -704,3 +704,39 @@ metric:
     expr: bookings
     metrics:
       - name: bookings
+---
+metric:
+  name: twice_bookings_fill_nulls_with_0_without_time_spine
+  description: 2x twice_bookings_fill_nulls_with_0_without_time_spine
```
Comment (tlento) [inline metricflow/test/fixtures/semantic_manifest_yamls/simple_manifest/metrics.yaml:710]: nit: `2x bookings_fill_nulls_with_0_without_time_spine`

File: metricflow/test/fixtures/semantic_manifest_yamls/simple_manifest/metrics.yaml [hunk 2]
```diff
@@ -704,3 +704,39 @@ metric:
...
+  type: derived
+  type_params:
+    expr: 2 * bookings_fill_nulls_with_0_without_time_spine
+    metrics:
+      - name: bookings_fill_nulls_with_0_without_time_spine
+---
+metric:
+  name: nested_fill_nulls_without_time_spine
+  description: 2x twice_bookings_fill_nulls_with_0_without_time_spine but then x3
```
Comment (tlento) [inline metricflow/test/fixtures/semantic_manifest_yamls/simple_manifest/metrics.yaml:719]: Same nit, but this time maybe strike the `2x `

File: metricflow/test/plan_conversion/dataflow_to_sql/test_conversion_metrics_to_sql.py [hunk 3]
```diff
@@ -168,3 +168,32 @@ def test_conversion_rate_with_constant_properties(
...
+
+@pytest.mark.sql_engine_snapshot
+def test_conversion_metric_fill_nulls_with_0(
+    request: FixtureRequest,
+    mf_test_session_state: MetricFlowTestSessionState,
+    dataflow_plan_builder: DataflowPlanBuilder,
+    dataflow_to_sql_converter: DataflowToSqlQueryPlanConverter,
+    sql_client: SqlClient,
+) -> None:
+    """Test conversion metric with constant properties by data flow plan rendering."""
```
Comment (tlento) [inline metricflow/test/plan_conversion/dataflow_to_sql/test_conversion_metrics_to_sql.py:181]: Let's update the docstring here and in the other test cases to indicate any weirdness. Like if this won't fill nulls at every level, we can put that in with a note that this should highlight the relevant changes if/when we alter that behavior.

File: metricflow/test/snapshots/test_conversion_metrics_to_sql.py/SqlQueryPlan/BigQuery/test_conversion_metric_fill_nulls_with_0__plan0_optimized.sql [hunk 4]
```diff
@@ -0,0 +1,84 @@
+-- Compute Metrics via Expressions
+SELECT
+  metric_time__day
+  , CAST(buys AS FLOAT64) / CAST(NULLIF(visits, 0) AS FLOAT64) AS visit_buy_conversion_rate_7days_fill_nulls_with_0
```
Comment (tlento) [inline metricflow/test/snapshots/test_conversion_metrics_to_sql.py/SqlQueryPlan/BigQuery/test_conversion_metric_fill_nulls_with_0__plan0_optimized.sql:4]: Oh interesting. If we want to support recursively filling nulls with something we'll need to apply it on the metric, because conversion and ratio metrics will return NULL for any zero denominator.

This is the default behavior we want - NULL is the correct result - but users may override when they know, for example, that any zero denominator implies a 0% conversion rate for their analysis purposes.

Nothing to do about that now except maybe make a note of this in the test docstring.

**Review by tlento — APPROVED** [review]
Seems reasonable!
